acceptance-tests/src/test/java/tech/pegasys/ethsigner/tests/hashicorpVault/HashicorpVaultDocker.java

```diff
@@ -0,0 +1,222 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.tests.hashicorpVault;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static tech.pegasys.ethsigner.tests.WaitUtils.waitFor;
+
+import java.io.ByteArrayOutputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.CreateContainerCmd;
+import com.github.dockerjava.api.command.CreateContainerResponse;
+import com.github.dockerjava.api.command.ExecCreateCmdResponse;
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import com.github.dockerjava.api.exception.NotFoundException;
+import com.github.dockerjava.api.exception.NotModifiedException;
+import com.github.dockerjava.api.model.Capability;
+import com.github.dockerjava.api.model.ExposedPort;
+import com.github.dockerjava.api.model.HostConfig;
+import com.github.dockerjava.api.model.PortBinding;
+import com.github.dockerjava.api.model.Ports;
+import com.github.dockerjava.api.model.Ports.Binding;
+import com.github.dockerjava.core.command.ExecStartResultCallback;
+import com.github.dockerjava.core.command.PullImageResultCallback;
+import com.github.dockerjava.core.command.WaitContainerResultCallback;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class HashicorpVaultDocker {
+
+  private static final Logger LOG = LogManager.getLogger();
+  private static final String HASHICORP_VAULT_IMAGE = "vault:latest";
+  private static final int DEFAULT_HTTP_PORT = 8200;
+  public static final String vaultToken = "token";
+  private static final String[] CREATE_ETHSIGNER_SIGNING_KEY_SECRET = {
+    "vault",
+    "kv",
+    "put",
+    "secret/ethsignerSigningKey",
+    "value=8f2a55949038a9610f50fb23b5883af3b4ecb3c3bb792cbcefbd1542c692be63"
+  };
+
+  private final DockerClient docker;
+  private final String vaultContainerId;
+
+  private int port;
+
+  public HashicorpVaultDocker(final DockerClient docker) {
+    this.docker = docker;
+    pullVaultImage();
+    this.vaultContainerId = createVaultContainer();
+  }
+
+  public void start() {
+    LOG.info("Starting Hashicorp Vault Docker container: {}", vaultContainerId);
+    docker.startContainerCmd(vaultContainerId).exec();
+
+    LOG.info("Querying for the Docker dynamically allocated vault port number");
+    final InspectContainerResponse containerResponse =
+        docker.inspectContainerCmd(vaultContainerId).exec();
+    final Ports ports = containerResponse.getNetworkSettings().getPorts();
+    port = httpRpcPort(ports);
+    LOG.info("Http port for Hashicorp Vault: {}", port);
+
+    try {
```

Comment:
Don't do it this way, apply the pattern used in `PantheonNode. `awaitStartupCompletion()` where a service is called with the `WaitUtil` taking care of failure ...as unlike this `sleep` it will only wait the minimum amount of time (i.e. what if it's available in one second, with your ten second sleep?)

Reply:
I've just noticed you have `awaitStartupCompletion()` on this class ...why then do you need this sleep at all?

Reply:
Good catch. Forgot to take that out once I had the await startup method!